programs/drift_vaults/src/state/tokenized_vault_depositor.rs

```diff
@@ -0,0 +1,574 @@
+use std::cell::RefMut;
+
+use crate::error::ErrorCode;
+use crate::events::{VaultDepositorAction, VaultDepositorRecord};
+use crate::state::vault::Vault;
+use crate::{validate, VaultFee, VaultProtocol};
+use crate::{Size, VaultDepositorBase};
+use static_assertions::const_assert_eq;
+
+use anchor_lang::prelude::*;
+use drift::math::casting::Cast;
+use drift::math::insurance::{
+    if_shares_to_vault_amount as depositor_shares_to_vault_amount,
+    vault_amount_to_if_shares as vault_amount_to_depositor_shares,
+};
+use drift::math::safe_math::SafeMath;
+use drift_macros::assert_no_slop;
+
+#[assert_no_slop]
+#[account(zero_copy(unsafe))]
+#[derive(Default, Eq, PartialEq, Debug)]
+#[repr(C)]
+pub struct TokenizedVaultDepositor {
+    /// The vault deposited into
+    pub vault: Pubkey,
+    /// The vault depositor account's pubkey. It is a pda of vault
+    pub pubkey: Pubkey,
+    /// The token mint for tokenized shares owned by this VaultDepositor
+    pub mint: Pubkey,
+    /// share of vault owned by this depositor. vault_shares / vault.total_shares is depositor's ownership of vault_equity
+    vault_shares: u128,
+    /// stores the vault_shares from the most recent liquidity event (redeem or issuance) before a spl token
+    /// CPI is done, used to track invariants
+    last_vault_shares: u128,
+    /// creation ts of vault depositor
+    pub last_valid_ts: i64,
+    /// lifetime net deposits of vault depositor for the vault
+    pub net_deposits: i64,
+
+    /// lifetime total deposits
+    pub total_deposits: u64,
+    /// lifetime total withdraws
+    pub total_withdraws: u64,
+    /// the token amount of gains the vault depositor has paid performance fees on
+    pub cumulative_profit_share_amount: i64,
+    pub profit_share_fee_paid: u64,
+    /// The exponent for vault_shares decimal places at the time the tokenized vault depositor was initialized.
+    /// If the vault undergoes a rebase, this TokenizedVaultDepositor can no longer issue new tokens, only redeem
+    /// is possible.
+    pub vault_shares_base: u32,
+    /// The bump for the vault pda
+    pub bump: u8,
+    pub padding: [u8; 3],
```

Comment:
might give it a bit more padding jsut in case

Reply:
think adding 5 bytes is enough?